module_utils/ansible_nailgun_cement.py

```diff
@@ -520,9 +520,12 @@ def find_content_credential(module, name, organization, failsafe=False):
     return handle_find_response(module, content_credential.search(), message="No content credential found for %s" % name, failsafe=failsafe)
 
 
-def find_repository_set(module, name, product, failsafe=False):
-    repo_set = RepositorySet(name=name, product=product)
-    return handle_find_response(module, repo_set.search(), message="No repository set found for %s" % name, failsafe=failsafe)
+def find_repository_set(module, name, product, organization=None, label=None, failsafe=False):
```

Comment:
IMO this should be:

```py
def find_repository_set(module, name, product=None, organization=None, failsafe=False)
```

Repo sets index API takes either organization ID or product ID (and repo set name) to resolve the repo set ID. I'm confused what this label is. Care to explain?

Reply:
I don't understand. I'm looking at our [API controller params](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/repository_sets_controller.rb#L23) and don't see label. It looks like you can search with either `org ID + name` or `product ID + name`.

Reply:
@akofink the label search uses scoped_search, which does allow us to search via label  https://github.com/Katello/katello/blob/master/app/models/katello/product_content.rb#L23

Reply:
Ah, ok that makes sense. :)

Reply:
So then, why is name required here in the method definition? You can use either name or label, right? So it should look like this:

```py
def find_repository_set(module, name=None, label=None, product=None, organization=None, failsafe=False)
```

This allows using name or label and product or organization. Maybe I'm missing something.

Reply:
Yup, nice spot! Updated :+1: Thanks!